pkg/controller/backup_controller.go

```diff
@@ -281,6 +288,57 @@ func (c *backupController) getLocationAndValidate(itm *api.Backup, defaultBackup
 	return backupLocation, validationErrors
 }
 
+// defaultAndValidateSnapshotLocations ensures:
+// - each location name in Spec VolumeSnapshotLocation exists as a location
+// - exactly 1 location per existing or default provider
+// - a given default provider's location name is added to the Spec VolumeSnapshotLocation if it does not exist as a VSL
+func (c *backupController) defaultAndValidateSnapshotLocations(itm *api.Backup, defaultLocations map[provider]string) []string {
+	var errors []string
+	perProviderLocationName := make(map[provider]string)
```

Comment:
not sure how much value the `provider` type alias is adding at this point - i'd probably just drop it. the var name makes it pretty clear what the key is.